fastavro/reader.py

```diff
@@ -214,6 +213,39 @@ def read_bytes_decimal(data, writer_schema=None, reader_schema=None):
     return scaled_datum
 
 
+def read_fixed_decimal(data, writer_schema=None, reader_schema=None):
```

Comment:
The only difference I see between this and `read_bytes_decimal` is `size = writer_schema['size']` vs `size = len(data)` (unless I'm missing something). Is there a way to define `size` that works for both and have the `LOGICAL_READERS` dictionary map both `bytes-decimal` and `fixed-decimal` to the same function?

Reply:
That is the only difference. I didn't like doing this, but didn't see an alternative without breaking the current coding patterns.

For example, we could do:
```
def read_fixed_decimal(data, writer_schema=None, reader_schema=None, size):
   ...

def read_bytes_decimal(data, writer_schema=None, reader_schema=None):
   size = len(data)
   read_fixed_decimal(data, writer_schema, reader_schema, size)
```

but adding the ```size``` param doesn't fit in with the other ```read_X``` functions.

Alternatively, we could use the current ``` read_bytes_decimal()``` for both bytes and fixed. It would work, but it if fixed got passed 6 bytes instead of 8 in the schema, say, it would still parse it .


I suppose we could do:
```
def read_decimal(data, writer_schema=None, reader_schema=None):
   if writer_schema:
      if writer_schema['type'] == 'fixed':
         size = writer_schema['size']
      elif  writer_schema['type'] == 'bytes
         size = len(data)
   ...
```

Reply:
What about something like this:

```python
def read_bytes_decimal(data, writer_schema=None, reader_schema=None):
    size = len(data)
    return _read_decimal(data, size, writer_schema)

def read_fixed_decimal(data, writer_schema=None, reader_schema=None):
    size = writer_schema['size']
    return _read_decimal(data, size, writer_schema)

def _read_decimal(data, size, writer_schema):
    scale = writer_schema['scale']
    precision = writer_schema['precision']

    datum_byte = str2ints(data)

    unscaled_datum = 0
    msb = fstint(data)
    leftmost_bit = (msb >> 7) & 1
    if leftmost_bit == 1:
        modified_first_byte = datum_byte[0] ^ (1 << 7)
        datum_byte = [modified_first_byte] + datum_byte[1:]
        for offset in xrange(size):
            unscaled_datum <<= 8
            unscaled_datum += datum_byte[offset]
        unscaled_datum += pow(-2, (size * 8) - 1)
    else:
        for offset in xrange(size):
            unscaled_datum <<= 8
            unscaled_datum += (datum_byte[offset])

    with localcontext() as ctx:
        ctx.prec = precision
        scaled_datum = Decimal(unscaled_datum).scaleb(-scale)
    return scaled_datum
```

It adds a function call, but reduces the duplication. Thoughts?

Reply:
Looks good to me.